src/Plugin/media/Source/Embeddable.php

```diff
@@ -0,0 +1,208 @@
+    $this->unstructuredField = $configuration['unstructured_field_name'];
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'oembed_field_name' => 'field_media_embeddable_oembed',
+      'unstructured_field_name' => 'field_media_embeddable_code',
+    ] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+    $options = $this->getSourceFieldOptions();
+    $form['oembed_field_name'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Field with oEmbed information'),
+      '#default_value' => $this->configuration['oembed_field_name'],
+      '#empty_option' => $this->t('-'),
+      '#options' => $options,
+      '#description' => $this->t('Select the field that will store the link, if it is an oEmbed.'),
+    ];
+    $form['unstructured_field_name'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Field for unstructured embed codes'),
+      '#default_value' => $this->configuration['unstructured_field_name'],
+      '#empty_option' => $this->t('-'),
+      '#options' => $options,
+      '#description' => $this->t('Select the field that will store essential information about the media item.'),
+    ];
+    return $form;
+  }
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getMetadata(MediaInterface $media, $name) {
+    if ($this->hasUnstructured($media)) {
+      return $this->getUnstructuredMetadata($media, $name);
+    }
+    return parent::getMetadata($media, $name);
+  }
+
+  /**
+   * Gets the value for a metadata attribute for a given media item.
+   *
+   * This is an alternate version to account for unstructured embeds.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   A media item.
+   * @param string $name
+   *   Name of the attribute to fetch.
+   *
+   * @return mixed|null
+   *   Metadata attribute value or NULL if unavailable.
+   */
+  public function getUnstructuredMetadata(MediaInterface $media, $name) {
+    // We short-circuit requests for URL here, to ensure kernel tests pass.
+    if (($name = 'url') && ($this->hasOEmbed($media))) {
+      return NULL;
+    }
+
+    switch ($name) {
+      case 'default_name':
+        if ($title = $this->getMetadata($media, 'title')) {
+          return $title;
+        }
+        return parent::getMetadata($media, 'default_name');
+
+      case 'thumbnail_uri':
+        return parent::getMetadata($media, 'thumbnail_uri');
+
+      case 'html':
+        return $media->get($this->unstructuredField)->getValue();
+
+      default:
```

Comment:
remove the `default` switch and the final return in this method. it will return `NULL` without a return.